Content.Server/_Moffstation/Objectives/Systems/HighValueTargetSelectionSystem.cs

```diff
@@ -0,0 +1,54 @@
+using Content.Server._Moffstation.Objectives.Components;
+using Content.Server.Preferences.Managers;
+using Content.Shared.Mind;
+using Content.Shared.Preferences;
+using Robust.Shared.Random;
+
+namespace Content.Server._Moffstation.Objectives.Systems;
+
+/// <summary>
+/// This handles selecting a high-value target for any systems that select random targets.
+/// </summary>
+public sealed class HighValueTargetSelectionSystem : EntitySystem
+{
+    [Dependency] private readonly IServerPreferencesManager _pref = default!;
+    [Dependency] private readonly IRobustRandom _random = default!;
+
+    /// <summary>
+    /// Selects a target from the list of possible targets with a priority on those with
+    /// the High Value Target role selected.
+    /// </summary>
+    public bool SelectTarget(
+        EntityUid rule,
+        HighValueTargetSelectionComponent? component,
+        List<Entity<MindComponent>> allPossibleTargets,
+        out Entity<MindComponent> target)
+    {
+        if (!Resolve(rule, ref component))
+        {
+            target = _random.Pick(allPossibleTargets);
+            return true;
+        }
+
+        _random.Shuffle(allPossibleTargets);
```

Comment:
To make this shuffle not mess with the list passed in, you make a defensive copy at the call site.
IMO the defensive copy should be made within this function so that callers need not have to remember they need to (unless you document the need, but even then, why document when you can enforce it here?)